pkg/login/auth.go

```diff
@@ -50,6 +50,7 @@ func AuthenticateUser(query *m.LoginUserQuery) error {
 
 	if err == ErrInvalidCredentials || err == LDAP.ErrInvalidCredentials {
 		saveInvalidLoginAttempt(query)
+		return ErrInvalidCredentials
```

Comment:
So we wouldn't want to propagate LDAP error?

Reply:
If we propagate LDAP error to the `LoginPost()` it will show different error (general "Error while trying to authenticate user" instead of "Invalid username or password"). This happens because of this comparison (`login.ErrInvalidCredentials != LDAP.ErrInvalidCredentials`)
https://github.com/grafana/grafana/blob/master/pkg/api/login.go#L104-L106